fido2/tpm.py

```diff
@@ -0,0 +1,455 @@
+        kdf_scheme = TpmiAlgKdf(reader.unpack("!H"))
+
+        return cls(symmetric, scheme, curve_id, kdf_scheme)
+
+    def __init__(self, symmetric, scheme, curve_id, kdf):
+        self.symmetric = symmetric
+        self.scheme = scheme
+        self.curve_id = curve_id
+        self.kdf = kdf
+
+    def __repr__(self):
+        return (
+            "<TpmsEccParms"
+            " symmetric=0x{self.symmetric:x}"
+            " scheme=0x{self.scheme:x}"
+            " curve_id={self.curve_id!r}"
+            " kdf={self.kdf!r}"
+            ">".format(self=self)
+        )
+
+
+class TpmsEccPoint(object):
+    """TPMS_ECC_POINT
+    https://www.trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-2-Structures-01.38.pdf
+    Section 11.2.5.2
+    """
+
+    @classmethod
+    def parse(cls, reader):
+        x = reader.read(reader.unpack("!H"))
+        y = reader.read(reader.unpack("!H"))
+
+        return cls(x, y)
+
+    def __init__(self, x, y):
+        self.x = y
+        self.y = y
+
+    def __repr__(self):
+        return "<TpmsEccPoint" " x={self.x}" " y={self.y}" ">".format(self=self)
+
+
+class TpmPublicFormat(object):
+    """the public area structure is defined by [TPMv2-Part2] Section 12.2.4 (TPMT_PUBLIC)
+    as:
+      TPMI_ALG_PUBLIC - type
+      TPMI_ALG_HASH - nameAlg
+        or + to indicate TPM_ALG_NULL
+      TPMA_OBJECT - objectAttributes
+      TPM2B_DIGEST - authPolicy
+      TPMU_PUBLIC_PARMS - type parameters
+      TPMU_PUBLIC_ID - uniq
+    See:
+      https://www.trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-2-Structures-01.38.pdf
+    """
+
+    class ATTRIBUTES(IntFlag):
+        """Object attributes
+        see section 8.3
+          https://www.trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-2-Structures-01.38.pdf
+        """
+
+        FIXED_TPM = 1 << 1
+        ST_CLEAR = 1 << 2
+        FIXED_PARENT = 1 << 4
+        SENSITIVE_DATA_ORIGIN = 1 << 5
+        USER_WITH_AUTH = 1 << 6
+        ADMIN_WITH_POLICY = 1 << 7
+        NO_DA = 1 << 10
+        ENCRYPTED_DUPLICATION = 1 << 11
+        RESTRICTED = 1 << 16
+        DECRYPT = 1 << 17
+        SIGN_ENCRYPT = 1 << 18
+
+        SHALL_BE_ZERO = (
+            (1 << 0)  # 0 Reserved
+            | (1 << 3)  # 3 Reserved
+            | (0x3 << 8)  # 9:8 Reserved
+            | (0xF << 12)  # 15:12 Reserved
+            | ((0xFFFFFFFF << 19) & (2 ** 32 - 1))  # 31:19 Reserved
+        )
+
+    @classmethod
+    def parse(cls, data):
+        reader = ByteBuffer(data)
+        sign_alg = TpmAlgAsym(reader.unpack("!H"))
+        hash_alg = TpmAlgHash(reader.unpack("!H"))
+
+        attributes = reader.unpack("!L")
+        if attributes & TpmPublicFormat.ATTRIBUTES.SHALL_BE_ZERO != 0:
+            raise ValueError(
+                "attributes is not formated correctly: 0x{:x}".format(attributes)
+            )
+        attributes = TpmPublicFormat.ATTRIBUTES(attributes)
```

Comment:
I _think_ this is the only line that would have to be removed if ATTRIBUTES change from an IntFlag to an IntEnum, everything else looks like it should work.